examples_tests/61.BlitFilterTest/default_compute_normalization.comp

```diff
@@ -0,0 +1,192 @@
+} pc;
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+nbl_glsl_blit_normalization_parameters_t nbl_glsl_blit_normalization_getParameters()
+{
+	return pc.params;
+}
+#define _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PADDED_DATA_DEFINED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ must be defined
+#endif
+
+nbl_glsl_blit_normalization_pixel_t nbl_glsl_blit_normalization_getPaddedData(in ivec3 coord)
+{
+	const nbl_glsl_blit_normalization_parameters_t params = nbl_glsl_blit_normalization_getParameters();
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#define COORD coord.x
+	#define IMAGE_DIM params.outImageDim.x
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#define COORD coord.xy
+	#define IMAGE_DIM params.outImageDim.xy
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#define COORD coord
+	#define IMAGE_DIM params.outImageDim
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+	nbl_glsl_blit_normalization_pixel_t result;
+	result.data = vec4(0.f);
+	if (all(lessThan(COORD, IMAGE_DIM)))
+	{
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		result.data = nbl_glsl_blit_formats_decode(imageLoad(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD));
+	#else
+		result.data = imageLoad(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD);
+	#endif
+	}
+
+#undef COORD
+#undef IMAGE_DIM
+
+	return result;
+}
+#define _NBL_GLSL_BLIT_NORMALIZATION_GET_PADDED_DATA_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_SET_DATA_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_SET_DATA_DEFINED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ must be defined
+#endif
+
+void nbl_glsl_blit_normalization_setData(in nbl_glsl_blit_normalization_pixel_t value, in ivec3 coord)
+{
+	const nbl_glsl_blit_normalization_parameters_t params = nbl_glsl_blit_normalization_getParameters();
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#define COORD coord.x
+	#define IMAGE_DIM params.outImageDim.x
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#define COORD coord.xy
+	#define IMAGE_DIM params.outImageDim.xy
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#define COORD coord
+	#define IMAGE_DIM params.outImageDim
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+	if (all(lessThan(COORD, IMAGE_DIM)))
+	{
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		imageStore(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD, nbl_glsl_blit_formats_encode(value.data));
+	#else
+		imageStore(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD, value.data);
+	#endif
```

Comment:
if you made a dummy (passthrough) `nbl_glsl_blit_formats_encode` available when not using software
```glsl
vec4 nbl_glsl_blit_encode(in vec4 data)
{
    return data;
}
```
you wouldn't need some of the ugly #ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_ throughout the code

Reply:
discord convo:
```glsl
#ifdef NBL_GLSL_BLIT_SOFTWARE_ENCODE
uvec4 nbl_glsl_blit_softwareEncode()
{
  uvec4 retval;
#if NBL_GLSL_BLIT_SOFTWARE_ENCODE==EF_R11G11B10_UFLOAT
#elif ...
#elif ...
#elif ...
#elif ...
#elif ...
#else
#error "Software Encode of this format not implemented or possible!"
#endif
  return retval;
}
#endif
```

Reply:
Resolved in the following commit.

Reply:
Pushed.